bin/pub_insights.dart

```diff
@@ -5,19 +5,69 @@ import 'dart:io';
 import 'dart:typed_data';
 
 import 'package:archive/archive_io.dart' as archive;
+import 'package:args/args.dart';
 import 'package:console_bars/console_bars.dart';
 import 'package:http/http.dart' as http;
 import 'package:path/path.dart' as path;
 import 'package:pub_insights/isolate_worker.dart';
 import 'package:pub_insights/pub_client.dart' as client;
 
+const String _directoryFlag = 'directory';
+const String _helpFlag = 'help';
+const String _fetchModeFlag = 'fetch-mode';
+const String _fetchModeAll = 'all';
+const String _fetchModePopular = 'popular';
+const String _fetchModeFile = 'file';
+const String _fetchModeNone = 'none';
+const String _fetchFileFlag = 'fetch-file';
+const String _optimizeTablesFlag = 'optimize-tables';
+
 Future<void> main(List<String> arguments) async {
-  if (arguments.length != 1) {
-    print('Usage: pub_insights <output path>');
-    return;
+  final ArgParser parser = ArgParser()
+    ..addOption(
+      _directoryFlag,
+      defaultsTo: 'data',
+      abbr: 'd',
+      help: 'Directory to download data into',
+    )
+    ..addOption(_fetchModeFlag,
+        allowed: ['all', 'popular', 'file', 'none'],
+        abbr: 'm',
+        help: 'What set of packages to download.',
+        defaultsTo: _fetchModePopular)
+    ..addOption(_fetchFileFlag,
+        abbr: 'f',
+        help:
+            'File to use as package input. Must be set if $_fetchModeFlag is file.')
+    ..addFlag(_optimizeTablesFlag,
+        abbr: 'o',
+        help: 'Collapse versions and stores into respective jsonl files',
+        defaultsTo: true)
+    ..addFlag(_helpFlag,
+        negatable: false, abbr: 'h', help: 'Print this reference.');
+  final ArgResults argResults = parser.parse(arguments);
+
+  if (argResults.flag(_helpFlag) || arguments.isEmpty) {
```

Comment:
Should we allow empty arguments? That should download popular packages to the `data` directory, right?

Reply:
Could you also update the [`README.md`'s instructions](https://github.com/loic-sharma/pub_insights/blob/c2a99428fa89684b03121994d2b55530cbd65d9d/README.md?plain=1#L17) to:

```
dart run ./bin/pub_insights.dart --mode popular
```

Reply:
Done